src/sinks/datadog/logs/sink.rs

```diff
@@ -102,6 +103,50 @@ fn normalize_event(event: &mut Event) {
         log.rename_key(host_path, event_path!("hostname"));
     }
 
+    if let Some(ddtags_path) = log.find_key_by_meaning("tags") {
+        let ddtags_path = ddtags_path.clone();
+
+        // first, if the value is an array we need to reconstruct it to a comma delimited string for DD logs intake.
+        if let Some(Value::Array(tags_arr)) = log.get(&ddtags_path) {
+            let n_tags = tags_arr.len();
+            if n_tags > 0 {
```

Comment:
… and then you don't need to remember `n_tags` and can use:
```suggestion
            if !tags_arr.is_empty() {
```